File: problemas/iniciante/1189/1189.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

let input = "";
let lineCount = 0;

process.stdin.on('data', (chunk) => {
    input += chunk;
});

process.stdin.on('end', () => {
    let lines = input.trim().split('\n');
    let O = lines.shift();

    let soma = 0.0;
    let count = 0;

    for (let i = 0; i < 12; ++i) {
        for (let j = 0; j < 12; ++j) {
            let valor = parseFloat(lines.shift());

            if ((j < 5) && ((i < 6 && j < i) || (i > 5 && j < 11 - i))) {
                soma += valor;
                ++count;
            }
        }
    }

    console.log(`${(O === 'S'? soma : soma / count).toFixed(1)}`)

});

File: problemas/iniciante/1189/1189.py
def main():
    O = input()
    soma = 0.0
    count = 0

    for i in range(12):
        for j in range(12):
            valor = float(input())

            if ((j < 5) and ((i < 6 and j < i) or (i > 5 and j < 11 - i))):
                soma += valor
                count += 1

    print(f"{soma}" if O == 'S' else f"{(soma / count):.1f}")

if __name__ == "__main__":
    main()
